core/input.js

```diff
@@ -1,23 +1,49 @@
+const KeyboardActionMapping = {
+  'ENTER': 13,
+  'KEY_A': 65,
+  'LEFT': 37,
+  'RIGHT': 39,
+  'UP': 38,
+  'DOWN': 40,
+};
+
 class Input {
   constructor() {
     this.pressed = {};
     this.events = {};
+    this._keyDownHandler = null;
+    this._keyUpHandler = null;
+  }
 
-    D.body.addEventListener('keydown', (e) => {
-      if (e.keyCode in this.pressed) {
-        e.preventDefault();
-        this.events[e.keyCode] = -1;
-        this.pressed[e.keyCode] = 1;
-      }
-    });
+  static build(targetElement) {
+    var input = new Input();
+    input.addEventListeners(targetElement);
+    input.bind(Object.values(KeyboardActionMapping));
+    return input;
+  }
 
-    D.body.addEventListener('keyup', (e) => {
-      if (e.keyCode in this.pressed) {
-        e.preventDefault();
-        this.events[e.keyCode] = 1;
-        this.pressed[e.keyCode] = 0;
-      }
-    });
+  keyUp(e) {
+    if (e.keyCode in this.pressed) {
+      e.preventDefault();
+      this.events[e.keyCode] = -1;
+      this.pressed[e.keyCode] = 1;
+    }
+  }
+
+  keyDown(e) {
+    if (e.keyCode in this.pressed) {
+      e.preventDefault();
+      this.events[e.keyCode] = 1;
+      this.pressed[e.keyCode] = 0;
+    }
+  }
+
+  addEventListeners(targetElement) {
+    this._keyDownHandler = this.keyDown.bind(this);
+    targetElement.addEventListener('keydown', this._keyDownHandler);
+
+    this._keyUpHandler = this.keyUp.bind(this);
+    targetElement.addEventListener('keyup', this._keyUpHandler);
```

Comment:
Same here
```javascript
targetElement.addEventListener('keydown', this.keyUp.bind(this));
```

Reply:
TL;DR: Yes, I think we can remove it.
Long answer: I added the to-do to unbind a long time ago but I can't remember exactly why. Trying to think about reasons I can come up with game mechanics that could require to ignore some key inputs for a given time but that's something we can achieve without unbind.